pages/index.tsx

```diff
@@ -1,9 +1,30 @@
-import Portal from '../components/graphics/portal';
-import {usePrivy} from '@privy-io/react-auth';
-import Head from 'next/head';
+import Portal from "../components/graphics/portal";
+import { usePrivy } from "@privy-io/react-auth";
+import { PrivyClient } from "@privy-io/server-auth";
+import { GetServerSideProps } from "next";
+import Head from "next/head";
+
+export const getServerSideProps: GetServerSideProps = async ({ req }) => {
+  const cookieAuthToken = req.cookies["privy-token"];
+  if (!cookieAuthToken) return { props: {} };
+
+  const PRIVY_APP_ID = process.env.NEXT_PUBLIC_PRIVY_APP_ID;
+  const PRIVY_APP_SECRET = process.env.PRIVY_APP_SECRET;
+  const client = new PrivyClient(PRIVY_APP_ID!, PRIVY_APP_SECRET!);
+
+  // Use this result to pass props to a page for server rendering or to drive redirects!
+  // ref https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props
+  const claims = await client.verifyAuthToken(cookieAuthToken);
+  console.log({ claims });
+
+  return {
+    props: {},
+    redirect: { destination: "/dashboard", permanent: false },
+  };
+};
```

Comment:
pages API was so much better at so many things when it make to stuff like this, nextjs middleware makes me sad

Reply:
tbh makes me wanna go back to developing actual servers using actual server frameworks

Reply:
You know I really agree, this is so easy to understand